pinot-core/src/main/java/org/apache/pinot/core/segment/creator/impl/inv/OffHeapBitmapInvertedIndexCreator.java

```diff
@@ -181,33 +188,51 @@ public void seal()
     }
 
     // Create bitmaps from inverted index buffers and serialize them to file
-    try (DataOutputStream offsetDataStream = new DataOutputStream(
-        new BufferedOutputStream(new FileOutputStream(_invertedIndexFile)));
-        FileOutputStream bitmapFileStream = new FileOutputStream(_invertedIndexFile);
-        DataOutputStream bitmapDataStream = new DataOutputStream(new BufferedOutputStream(bitmapFileStream))) {
-      int bitmapOffset = (_cardinality + 1) * Integer.BYTES;
-      offsetDataStream.writeInt(bitmapOffset);
-      bitmapFileStream.getChannel().position(bitmapOffset);
-
+    ByteBuffer offsetBuffer = null;
+    ByteBuffer bitmapBuffer = null;
+    try (FileChannel channel = new RandomAccessFile(_invertedIndexFile, "rw").getChannel()) {
+      // map the offsets buffer
+      final int startOfBitmaps = (_cardinality + 1) * Integer.BYTES;
+      int bitmapOffset = startOfBitmaps;
+      offsetBuffer = channel.map(FileChannel.MapMode.READ_WRITE, 0, bitmapOffset).order(LITTLE_ENDIAN);
+      offsetBuffer.putInt(reverseBytes(bitmapOffset));
+      RoaringBitmap[] bitmaps = new RoaringBitmap[_cardinality];
+      RoaringBitmapWriter<RoaringBitmap> writer = RoaringBitmapWriter.writer()
+              .initialCapacity(((_nextDocId - 1) >>> 16) / _cardinality).get();
```

Comment:
Why do you need to divide it with `_cardinality`? Though this should be `.expectedRange(0, _nextDocId)`?

Reply:
This is just a heuristic that the keys of the bitmap will be evenly distributed across the dictionary ids. It won't always be perfect, but is never that bad. I can provide more justification.

Reply:
From my understanding, this method preallocate the arrays based on the size of the prefix keys. Since we are going to reuse this writer for all the bitmaps, we will always end up using all the prefix keys from `(0, _nextDocId)`, and we should use `.expectedRange(0, _nextDocId)` to preallocate all of them. Maybe I missed something here?

Reply:
I felt this heuristic was distracting so removed it. 

In case this comes up in the future, expecting the range to be `[0, _nextDocId)` would allocate far too much memory for datasets sorted by the indexed dimension, but would be the right choice if the association between row id and dictionary id were uniformly random. Some metadata could probably be associated with each dictionary id (number of documents, min, max, density?) in memory to use these methods could likely be used more effectively, and not in a way that overfits to a benchmark.